src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/SymbolTable.cs

```diff
@@ -346,37 +346,17 @@ private List<Type> CreateInheritanceHierarchyList(Type type)
 
         private Name GetName(string p)
         {
-            if (p == null)
-            {
-                p = string.Empty;
-            }
-            return GetName(p, _nameManager);
+            return _nameManager.Add(p ?? "");
         }
 
         /////////////////////////////////////////////////////////////////////////////////
 
         private Name GetName(Type type)
         {
             string name = type.Name;
-            if (type.IsGenericType)
-            {
-                // Trim the name to remove the ` at the end.
-                name = name.Split('`')[0];
-            }
-            return GetName(name, _nameManager);
+            return type.IsGenericType ? _nameManager.Add(name, name.IndexOf('`')) : _nameManager.Add(name);
```

Comment:
The whole GetName deal looks a bit dubious. Is seems we are just mapping a string to the same string + null is mapped to "" and generic type names are stripped of the ` 'N  ` suffix.

Current changes look ok, since this is mostly refactoring, but I think the whole GetName deal can be replaced by a small cache for mapping generic names to the truncated form.